scripts/stage_layer.py

```diff
@@ -0,0 +1,94 @@
+from argparse import ArgumentParser
+import logging
+from pathlib import Path
+import shutil
+import subprocess
+import sys
+from zipfile import (
+    ZipFile,
+    ZipInfo,
+)
+
+import boto3
+
+from azul import config
+from azul.logging import configure_script_logging
+
+log = logging.getLogger(__name__)
+
+
+class LayerBuilder:
+    layer_dir = Path(config.project_root) / 'lambdas' / 'layer'
+    out_dir = layer_dir / '.chalice' / 'terraform'
+
+    def __init__(self):
+        self.client = boto3.client('s3')
+
+    def update_layer_if_necessary(self):
+        log.info('Updating layer in bucket: %s with key %s', config.layer_bucket, config.layer_object_key)
+        if self.update_required():
+            log.info('Updating layer.zip')
+            self.update_layer()
+        else:
+            log.info('No update necessary')
+
+    def update_required(self) -> bool:
+        log.info('Checking if layer.zip needs to be updated')
+        try:
+            # Since the object key is content-addressable, just checking for
+            # the file's presence is sufficient
+            self.client.head_object(Bucket=config.layer_bucket, Key=config.layer_object_key)
+        except self.client.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                return True
+            else:
+                raise
+        else:
+            return False
+
+    def update_layer(self, force: bool = False):
+        input_zip = self.out_dir / 'deployment.zip'
+        layer_zip = self.out_dir / 'layer.zip'
+        if force:
+            log.info('Tainting old lambda layer')
+            command = ['terraform', 'taint', 'aws_lambda_layer_version.dependencies']
+            subprocess.run(command, cwd=Path(config.project_root) / 'terraform')
+        self.build_layer_zip(layer_zip, input_zip)
+        log.info('Uploading layer zip to S3')
```

Comment:
```suggestion
        log.info('Uploading layer ZIP to S3')
```

https://en.wikipedia.org/wiki/Zip_(file_format)